### ISSUE TITLE
False positive for filter_var()

### ISSUE THREAD

**Issue by rybakit** [vimeo/psalm#2780]
The `FILTER_NULL_ON_FAILURE` option is not taken into account when checking for integers:

```php
null === $value = filter_var($var, FILTER_VALIDATE_INT, FILTER_NULL_ON_FAILURE);
```
Psalm (3.8.5@e6ec5fa22a7b9e61670a24d07b3119aff80dcd89) complains that the `$value` can only be `false|int`, though it's actually `null|int`.

Snippet: https://psalm.dev/r/4a4fcadaeb.
